config/config.go

```diff
@@ -100,34 +84,57 @@ type PeerPortConfig struct {
+	MonitorPeers                            bool                                 `mapstructure:"monitorPeers"`
+	CheckUtxos                              bool                                 `mapstructure:"checkUtxos"`
+	Health                                  *HealthConfig                        `mapstructure:"health"`
+	RejectCallbackContaining                []string                             `mapstructure:"rejectCallbackContaining"`
+	Stats                                   *StatsConfig                         `mapstructure:"stats"`
+	BlockchainNetwork                       *BlockchainNetwork[*MetamorphGroups] `mapstructure:"bcnet"`
 }
 
 type BlocktxConfig struct {
-	ListenAddr                    string              `mapstructure:"listenAddr"`
-	DialAddr                      string              `mapstructure:"dialAddr"`
-	HealthServerDialAddr          string              `mapstructure:"healthServerDialAddr"`
-	Db                            *DbConfig           `mapstructure:"db"`
-	RecordRetentionDays           int                 `mapstructure:"recordRetentionDays"`
-	RegisterTxsInterval           time.Duration       `mapstructure:"registerTxsInterval"`
-	MaxBlockProcessingDuration    time.Duration       `mapstructure:"maxBlockProcessingDuration"`
-	MonitorPeers                  bool                `mapstructure:"monitorPeers"`
-	FillGaps                      *FillGapsConfig     `mapstructure:"fillGaps"`
-	MaxAllowedBlockHeightMismatch int                 `mapstructure:"maxAllowedBlockHeightMismatch"`
-	MessageQueue                  *MessageQueueConfig `mapstructure:"mq"`
-	P2pReadBufferSize             int                 `mapstructure:"p2pReadBufferSize"`
+	ListenAddr                    string                             `mapstructure:"listenAddr"`
+	DialAddr                      string                             `mapstructure:"dialAddr"`
+	HealthServerDialAddr          string                             `mapstructure:"healthServerDialAddr"`
+	Db                            *DbConfig                          `mapstructure:"db"`
+	RecordRetentionDays           int                                `mapstructure:"recordRetentionDays"`
+	RegisterTxsInterval           time.Duration                      `mapstructure:"registerTxsInterval"`
+	MaxBlockProcessingDuration    time.Duration                      `mapstructure:"maxBlockProcessingDuration"`
+	MonitorPeers                  bool                               `mapstructure:"monitorPeers"`
+	FillGaps                      *FillGapsConfig                    `mapstructure:"fillGaps"`
+	MaxAllowedBlockHeightMismatch int                                `mapstructure:"maxAllowedBlockHeightMismatch"`
+	MessageQueue                  *MessageQueueConfig                `mapstructure:"mq"`
+	P2pReadBufferSize             int                                `mapstructure:"p2pReadBufferSize"`
+	BlockchainNetwork             *BlockchainNetwork[*BlocktxGroups] `mapstructure:"bcnet"`
```

Comment:
I think this is unnecessarily complicated. You could just create 2 structs, such as `BlocktxMcast` and `MetamorphMcast` or something like that. Would be easier to use it later in code as well. But it's just my opinion :)